packages/vaex-ml/vaex/ml/sklearn.py

```diff
@@ -95,3 +95,141 @@ def fit(self, df, target, **kwargs):
+        return self.model.predict(X)
+
+    def predict(self, df):
+        '''Get an in-memory numpy array with the predictions of the SKLearnPredictor.self
+
+        :param df: A vaex DataFrame, containing the input features.
+        :returns: A in-memory numpy array containing the SKLearnPredictor predictions.
+        :rtype: numpy.array
+        '''
+        data = df[self.features].values
+        return self.model.predict(data)
+
+    def transform(self, df):
+        '''Transform a DataFrame such that it contains the predictions of the IncrementalPredictor.
+        in form of a virtual column.
+
+        :param df: A vaex DataFrame.
+
+        :return copy: A shallow copy of the DataFrame that includes the IncrementalPredictor prediction as a virtual column.
+        :rtype: DataFrame
+        '''
+        copy = df.copy()
+        lazy_function = copy.add_function('incremental_prediction_function', self, unique=True)
+        expression = lazy_function(*self.features)
+        copy.add_virtual_column(self.prediction_name, expression, unique=False)
+        return copy
+
+    def fit(self, df, target, **kwargs):
+        '''Fit the IncrementalPredictor to the DataFrame.
+
+        :param df: A vaex DataFrame containing the features on which to train the model.
+        :param target: The name of the column containing the target variable.
+        '''
+        # Check whether the model is appropriate
+        assert hasattr(self.model, 'partial_fit'), 'The model must have a `.partial_fit` method.'
+
+        # Number of batches to send (or training rounds)
+        num_batches = np.int(np.ceil(len(df) / self.batch_size))
+
+        # For slicing the df
+        index_min = 0
+        index_max = self.batch_size
+
+        # For reference: tqdm version was - would be nice to have an informative progress bar
+        # for i in tqdm(range(num_batches), leave=True, desc='Training in batches...'):
+
+        # Main loop: pass on batches to the model
+        for batch in range(num_batches):
+            # Get data for this batch
+            X = df[index_min:index_max][self.features].values
+            y = df[index_min:index_max][target].values
+            shuffle_index = np.arange(len(X))
+
+            # For reference: tqdm version was - would be nice to have an informative progress bar
+            # for j in tqdm(range(num_epochs), leave=False, desc='Iterating over batch...'):
+
+            # Loop over for each epoch
+            for epoch in range(self.num_epochs):
+                if self.shuffle:
+                    shuffle_index = np.arange(len(X))
+                    np.random.shuffle(shuffle_index)
+                    X = X[shuffle_index]
+                    y = y[shuffle_index]
+
+                # train the model
+                self.model.partial_fit(X, y)
+
+            # update the slicing indices to be ready for the next batch
+            index_min += self.batch_size
+            index_max += self.batch_size
```

Comment:
```suggestion
            index_max += self.batch_size
            index_max = min(N_total, index_max)  # clip upper value
```
Is this not triggered by the test?